examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart

```diff
@@ -68,15 +66,15 @@ class _DatePickerExampleState extends State<DatePickerExample> {
           fontSize: 22.0,
         ),
         child: Center(
-          child: Column(
-            mainAxisAlignment: MainAxisAlignment.center,
-            children: <Widget>[
-              _DatePickerItem(
-                children: <Widget>[
-                  const Text('Date'),
-                  CupertinoButton(
-                    // Display a CupertinoDatePicker in date picker mode.
-                    onPressed: () => _showDialog(
+          child: Column(mainAxisAlignment: MainAxisAlignment.center, children: <
+            Widget
+          >[
+            _DatePickerItem(children: <Widget>[
```

Comment:
This looks odd.

Reply:
Haha, this one *is* odd. I think the long line comments inside the closure below are tripping it up. It's desparately trying to fit them in 80 columns, so it picks a weird formatting for the outer call to minimize indentation. I think if you rewrap those comments to be shorter, it will do something sane here.

Reply:
I deleted all the comments in this block and ran the formatter again. The output was exactly the same. Not sure how to "trick" the formatter here to do something nicer. Ultimately, this appears to be another instance of https://github.com/dart-lang/dart_style/issues/1527, where the argument list of Column should really be using trailing-comma style to avoid this oddity.

Reply:
OK, I looked into it. In this case, it was the long string interpolation line. If you split that, then you get better output:

```dart
  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      navigationBar: const CupertinoNavigationBar(
        middle: Text('CupertinoDatePicker Sample'),
      ),
      child: DefaultTextStyle(
        style: TextStyle(
          color: CupertinoColors.label.resolveFrom(context),
          fontSize: 22.0,
        ),
        child: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              _DatePickerItem(children: <Widget>[
                ...
              ]),
              ...
              _DatePickerItem(children: <Widget>[
                const Text('DateTime'),
                CupertinoButton(
                  // Display a CupertinoDatePicker in dateTime picker mode.
                  onPressed: ...
                  // In this example, the time value is formatted manually. You
                  // can use the intl package to format the value based on the
                  // user's locale settings.
                  child: Text(
                    '${dateTime.month}-${dateTime.day}-${dateTime.year} '
                    '${dateTime.hour}:${dateTime.minute}',
                    style: const TextStyle(fontSize: 22.0),
                  ),
                ),
              ]),
            ],
          ),
        ),
      ),
    );
  }
```

I agree that https://github.com/dart-lang/dart_style/issues/1527 would help here, as would softer overflow handling rules. There is also https://github.com/dart-lang/dart_style/issues/1516. I was thinking about that last night and one way to tackle it might be to allow overflow if it's caused by string interpolations or other unsplittable constructs. So there might be a few ways to tackle this.